_posts/2020-09-28-biased-locking-help.adoc

```diff
@@ -0,0 +1,88 @@
+---
+layout: post
+title: Biased locking removed from Java - does it affect you ?
+date: 2020-09-28
+tags: help
+synopsis: In Java 15 *biased locking* is now disabled by default and is planned to be removed in the Java Virtual Machine as per JEP 374. We need your help seeing if it affects Quarkus performance in your usecase.
+author: maxandersen
+---
+
+Last week we became aware that the OpenJDK team in Java 15 have disabled so called *biased locking* (https://openjdk.java.net/jeps/374[JEP 374]) in the Java virtual machine. This is change from previous years and could *potentially* have a negative impact on Java application's performance.
+
+Red Hat's own performance teams are currently running performance tests to see how it affects our Java middleware, but no amount of generic testing can reveal how this change affects real world usage.
+
+That is where you come in.
+
+We would like to get info from you on whether your application performance is affected by biased locking or not.
+
+To do so please try the following in your application performance tests:
+
+Run your Quarkus application tests/performance as you normally would with the following command line flags on Java 11 (jdk11u) and if feasible Java 15 (jdk15):
+
+*enabled*: -XX:+UseBiasedLocking -XX:BiasedLockingStartupDelay=0
+
+*disabled*: -XX:-UseBiasedLocking
+
+We would like to know the result of this no matter if you see a regression or not on the same Java Virtual Machine.
+
+If you are able then running it with the following scenarios would be of great help too:
+
+ . single-threaded
+ . thread count ~= hardware core count
+ . thread count ~= N * hardware core count where 8 < N < 16
+
+ The intent of these are to see how the level of concurrency affects the result.
+
+=== Reporting the result
+
+Please open a bug on with `[jep374]` in the title + your project. i.e. `[jep374] results for acme project crazy panda` containing the following information per run in the description:
+
+```
+jvm used: jdk11 or jdk15
+biased locking: enabled or disabled
+thread-count: 1..N (if you know)
+hardware-core count: N (if you know)
```

Comment:
In order to quantify any impact, we need to see the results with and without biased locking enabled

```suggestion
hardware-core count: N (if you know)
performance test result: with biased locking
performance test result: without biased locking
```